src/dll/Systems/ScriptSystem.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "Hook.hpp"
+#include "ISystem.hpp"
+#include "Paths.hpp"
+#include "PluginBase.hpp"
+
+class ScriptSystem : public ISystem
+{
+public:
+    ScriptSystem(const Paths& aPaths);
+
+    ESystemType GetType() final;
+
+    void Startup() final;
+    void Shutdown() final;
+
+    bool Add(std::shared_ptr<PluginBase> aPlugin, const char* path);
+    std::wstring GetRedModArgs();
+    void SetScriptsBlobPath(RED4ext::CString*);
```

Comment:
Should this be a string or a std::filesystem::path?

Reply:
We're not doing anything with it, filesystem-wise, so I was just using the type it was in the Red struct.

Reply:
I still think it is better to be a std::filesystem::path even if we are not doing anything with it. It is easier to understand when quickly looking at it, even if the name says path.